### ISSUE TITLE
Create a mecanism to turn the github api on/off using environmental variables. Please be quick!

### ISSUE THREAD

**Issue by sheenarbw** [Umuzi-org/Tilde#48]
Currently, for a dev to be able to get everything running on their local machine and run through all card interactions, they need to have a github api key, and they need to have authenticated with a github user that acts on behalf of the platform and all sorts of complicated painful things.

What we need is to make it easy for devs to dev with a simple and easy setup.

When a dev is working on their local machine, if they have the gui running and they click the "start project" card then it should appear to work - a Repository instance should be created in the db, the card should move, all the good stuff.   But we need to have a way to turn off creation of actual repos.

So basically we need a way to turn mocking of github api calls on and off.

This needs to be controlled via environmental variables.
The default behavior should be that the github interactions are turned off

If github interactions are being mocked then every time there would usually be a github interaction, log that it has been suppressed. This should be done just to inform the dev so that they dont get surprised when the repos dont exist. Try to  minimise possible confusion. Note that many users of this sstem are not english-first language speakers, and many will be junior. So clarity!
